dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/AbstractJdbcEventAnalyticsManager.java

```diff
@@ -1153,6 +1151,41 @@ protected String getQueryItemsAndFiltersWhereClause(EventQueryParams params, Sql
         + joinSql(Stream.concat(orConditions.stream(), andConditions.stream()), AND_JOINER);
   }
 
+  /**
+   * @param relation the relation to quote, e.g. a table or column name.
+   * @return a double quoted relation.
+   */
+  protected String quote(String relation) {
+    return sqlBuilder.quote(relation);
+  }
+
+  /**
+   * @param relation the relation to quote.
+   * @return an "ax" aliased and double quoted relation.
+   */
+  protected String quoteAlias(String relation) {
+    return sqlBuilder.quoteAx(relation);
+  }
+
+  /**
+   * @param value the value to quote.
+   * @return a single quoted value.
+   */
+  protected String singleQuote(String value) {
+    return sqlBuilder.singleQuote(value);
+  }
+
+  /**
+   * Returns a concatenated string of the given items separated by comma where each item is quoted
+   * and aliased.
+   *
+   * @param items the collection of items.
+   * @return a string.
+   */
+  protected String quoteAliasCommaDelimited(Collection<String> items) {
+    return items.stream().map(this::quoteAlias).collect(Collectors.joining(","));
+  }
```

Comment:
isn't it mostly duplicated code of `dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/data/JdbcAnalyticsManager.java:1018` ?

Reply:
Yes agreed. Just didn't want to introduce a huge diff or add yet another abstract class.